docs/auth.md

```diff
@@ -22,64 +21,108 @@ The authentication system in Soul aims to be very generic and doesn't provide so
+Due to the sensitive nature of the superuser status, it is not possible to change the `is_superuser` attribute of a user through the API.
 
-Creating superusers:
+Instead you can create, update and delete superusers using the command line.
 
-Create superusers using the createsuperuser command:
+Create superusers using the `createsuperuser` command:
 
 ```
 $ node src/server.js createsuperuser --username=joe --password=strongstring
 ```
 
-#### Permissions
+Updating superusers using the `updatesuperuser` command:
 
-Soul comes with a built-in permissions system. It provides a way to assign permissions to specific users and groups of users.
+```
+$ node src/server.js updatesuperuser --username=joe --password=newstrongstring
+```
 
-Soul uses permissions as follows:
+Deleting superusers using the `updatesuperuser` command:
 
-- Access to view objects is limited to users with the "read" permission for that type of object.
-- Access to add an object is limited to users with the "create" permission for that type of object.
-- Access to change an object is limited to users with the "update" permission for that type of object.
-- Access to delete an object is limited to users with the "delete" permission for that type of object.
+```
+$ node src/server.js deletesuperuser --username=joe
 
-Default Permissions:
+```
 
-By default, Soul ensures that four default permissions – create, update, delete, and read – are created for each table defined in one of your databases.
+#### Roles
 
-| table_name | can_create | can_read | can_update | can_delete |
-| ---------- | ---------- | -------- | ---------- | ---------- |
-| albums     | false      | true     | true       | false      |
+Roles are a generic way of categorizing users so you can assign permissions to those users. A user can belong to any number of roles.
 
-In this scenario, calling POST (create) and GET (read) requests to /tables/albums/rows is allowed by any user (even anonymous), while PUT (update) and DELETE (delete, of course) requests are not allowed.
+A user with a role automatically has the permissions granted to that role. For example, if the role "editor" has the permission `can_update_posts`, any user with that role will have that permission.
 
-When Soul boots up, it checks for the existence of a table called \_default_permissions.
+The attributes of the `_roles` table are:
 
-#### Roles
+- id int
+- name varchar
+- created_at datetime
+- updated_at datetime
+
+##### Roles Permissions
+
+Using a table called `_roles_permissions` we can assign permissions to roles.
+
+The attributes of this table are:
+
+- id int
+- role_id int
+- table_name varchar
+- create boolean
+- read boolean
+- update boolean
+- delete boolean
+
+There is unique constraint on the combination of `role_id`, `table_name`, `create`, `read`, `update` and `delete` attributes,
+to prevent duplication of permissions for the same role and table.
```

Comment:
Isn't it a constraint on `role_id` and `table` being unique?  If the `read` values are different there won't be a uniqueness clash, and we'll accidentally allow two sets of permissions for one role/table.

Reply:
Sure, will update it.